alica_tests/src/test/test_config_change.cpp

```diff
@@ -0,0 +1,153 @@
+{
+    ros::NodeHandle nh;
+    std::string path;
+    nh.param<std::string>("/rootPath", path, ".");
+
+    AlicaContext *ac = new alica::AlicaContext("RoleSet", "MasterPlan", true, path + "/etc/Alica_nase.yaml");
+    ac->setLocalAgentName("nase");
+
+    ASSERT_TRUE(ac->isValid());
+    ac->setCommunicator<alicaDummyProxy::AlicaDummyCommunication>();
+    alica::AlicaCreators creators(std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
+                                  std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>());
+    AlicaEngine* ae = AlicaTestsEngineGetter::getEngine(ac);
+    const_cast<IAlicaCommunication&>(ae->getCommunicator()).startCommunication();
+
+    EXPECT_TRUE(ac->setOption<bool>("Alica.SilentStart", false));
+    EXPECT_TRUE(ae->maySendMessages());
+
+    EXPECT_TRUE(ac->setOption<bool>("Alica.SilentStart", true));
+    EXPECT_FALSE(ae->maySendMessages());
+
+    EXPECT_TRUE(ac->setOption<bool>("Alica.AllowIdling", false));
+    EXPECT_FALSE(PartialAssignment::isIdlingAllowed());
+
+    EXPECT_TRUE(ac->setOption<bool>("Alica.AllowIdling", true));
+    EXPECT_TRUE(PartialAssignment::isIdlingAllowed());
+
+    EXPECT_TRUE(ac->setOption<std::string>("Local.DefaultRole", "ExampleRole"));
+    EXPECT_EQ("ExampleRole", ae->getTeamManager().getLocalAgent()->getProperties().getDefaultRole());
+
+    EXPECT_TRUE(ac->setOption<std::string>("Local.DefaultRole", "Attacker"));
+    EXPECT_EQ("Attacker", ae->getTeamManager().getLocalAgent()->getProperties().getDefaultRole());
+
+    EXPECT_TRUE(ae->init(creators));
+
+    ac->terminate();
+    delete ac;
+}
+
+    TEST(ConfigUpdates, TestBlockConfigUpdatesAfterInitialization)
+{
+    ros::NodeHandle nh;
+    std::string path;
+    nh.param<std::string>("/rootPath", path, ".");
+
+    AlicaContext *ac = new alica::AlicaContext("RoleSet", "MasterPlan", true, path + "/etc/Alica_nase.yaml");
+    ac->setLocalAgentName("nase");
+
+    ASSERT_TRUE(ac->isValid());
+    ac->setCommunicator<alicaDummyProxy::AlicaDummyCommunication>();
+    alica::AlicaCreators creators(std::make_unique<alica::ConditionCreator>(), std::make_unique<alica::UtilityFunctionCreator>(),
+                                  std::make_unique<alica::ConstraintCreator>(), std::make_unique<alica::BehaviourCreator>());
+    AlicaEngine* ae = AlicaTestsEngineGetter::getEngine(ac);
+    const_cast<IAlicaCommunication&>(ae->getCommunicator()).startCommunication();
+
+    ac->setOption<int>("Alica.TeamTimeOut", 1000);
+    EXPECT_EQ(1000, ac->getConfig()["Alica"]["TeamTimeOut"].as<int>());
+
+    EXPECT_TRUE(ae->init(creators));
+
+    //changes to config not allowed after initialization of AlicaContext
+    ASSERT_FALSE(ac->setOption<int>("Alica.TeamTimeOut", 2000));
+    EXPECT_EQ(1000, ac->getConfig()["Alica"]["TeamTimeOut"].as<int>());
+
+    ac->terminate();
+    delete ac;
+}
+
+    TEST(ConfigUpdates, TestConfigUpdatesWithVector)
```

Comment:
a lot of code for setup and termination is duplicated in every test. You can have an AlicaConfigTestFixture similar to AlicaTestFixtureBase and have all tests inherit from it